core/core.go

```diff
@@ -538,6 +567,22 @@ func (t *Textile) noAuthFunc(ctx context.Context) (context.Context, error) {
 	return ctx, nil
 }
 
+func (t *Textile) powergateInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+	if !strings.Contains(info.FullMethod, "ffs.rpc.RPCService") {
+		return handler(ctx, req)
+	}
+	methodParts := strings.Split(info.FullMethod, "/")
+	if len(methodParts) < 2 {
+		return nil, fmt.Errorf("error parsing method string %s", info.FullMethod)
+	}
+	methodName := methodParts[len(methodParts)-1]
+	methodDesc := t.ffsServiceDesc.FindMethodByName(methodName)
+	if methodDesc == nil {
+		return nil, fmt.Errorf("no method found for %s", methodName)
+	}
+	return t.powStub.InvokeRpc(ctx, methodDesc, req.(proto.Message))
```

Comment:
Use our stub to invoke the method on the remote powergate server. Boom, dynamic gRPC.